src/main/java/com/polyscripts/contactManagement/models/Contact.java

```diff
@@ -0,0 +1,25 @@
+package com.polyscripts.contactManagement.models;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.persistence.*;
+import java.io.Serializable;
+import java.util.List;
+
+@Entity
+@Getter @Setter
+@DiscriminatorColumn(name = "Type_Contact", discriminatorType = DiscriminatorType.STRING)
+public abstract class Contact implements Serializable {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(nullable = false, updatable = false)
+    private Long id;
+    private String name;
+    private String lastname;
+    private String address;
+
+    public abstract List<Enterprise> getEnterprises();
```

Comment:
why abstract?
use Set instead
where is relationship here ?
you just add a getter annotation, so no needs to declare getter here